test/algorithms/test_effective_dimension.py

```diff
@@ -0,0 +1,209 @@
+
+import unittest
+from test import QiskitMachineLearningTestCase
+
+import numpy as np
+from ddt import ddt, data
+
+from qiskit import Aer, QuantumCircuit
+from qiskit.circuit.library import ZFeatureMap, RealAmplitudes
+from qiskit.utils import QuantumInstance, algorithm_globals
+from qiskit.utils import optionals
+
+from qiskit.opflow import PauliSumOp
+from qiskit_machine_learning.neural_networks import TwoLayerQNN, CircuitQNN
+from qiskit_machine_learning.algorithms.effective_dimension import (
+    EffectiveDimension,
+    LocalEffectiveDimension,
+)
+
+
+@ddt
+class TestEffDim(QiskitMachineLearningTestCase):
+    """Test the Effective Dimension algorithm"""
+
+    def setUp(self):
+        super().setUp()
+
+        # fix seeds
+        algorithm_globals.random_seed = 1234
+        qi_sv = QuantumInstance(
+            Aer.get_backend("aer_simulator_statevector"),
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+
+        # set up quantum neural networks
+        num_qubits = 3
+        feature_map = ZFeatureMap(feature_dimension=num_qubits, reps=1)
+        ansatz = RealAmplitudes(num_qubits, reps=1)
+
+        # CircuitQNNs
+        qc = QuantumCircuit(num_qubits)
+        qc.append(feature_map, range(num_qubits))
+        qc.append(ansatz, range(num_qubits))
+
+        def parity(x):
+            return f"{x:b}".count("1") % 2
+
+        self.circuit_qnn_1 = CircuitQNN(
+            qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            interpret=parity,
+            output_shape=2,
+            sparse=False,
+            quantum_instance=qi_sv,
+        )
+
+        # qnn2 for checking result without parity
+        self.circuit_qnn_2 = CircuitQNN(
+            qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            sparse=False,
+            quantum_instance=qi_sv,
+        )
+
+        # OpflowQNN
+        observable = PauliSumOp.from_list([("Z" * num_qubits, 1)])
+        self.opflow_qnn = TwoLayerQNN(
+            num_qubits,
+            feature_map=feature_map,
+            ansatz=ansatz,
+            observable=observable,
+            quantum_instance=qi_sv,
+        )
+
+        # define sample numbers
+        self.n_list = [5000, 8000, 10000, 40000, 60000, 100000, 150000, 200000, 500000, 1000000]
+        self.n = 5000
+
+    @data(
+        # num_inputs, num_params
+        ("circuit1", 10, 10, 4.62355184),
+        ("circuit1", 1, 1, 1.39529449),
+        ("circuit1", 10, 1, 4.92825034),
+        ("circuit2", 10, 10, 5.93064171),
+    )
+    @unittest.skipUnless(optionals.HAS_AER, "qiskit-aer is required to run this test")
+    def test_alg_results(self, config):
+        """Test that the algorithm results match the original code's."""
+
+        qnn_name, num_inputs, num_params, result = config
```

Comment:
```suggestion
    @data(
        # num_inputs, num_params
        ("circuit1", 10, 10, 4.62355184),
        ("circuit1", 1, 1, 1.39529449),
        ("circuit1", 10, 1, 4.92825034),
        ("circuit2", 10, 10, 5.93064171),
    )
    @unpack
    @unittest.skipUnless(optionals.HAS_AER, "qiskit-aer is required to run this test")
    def test_alg_results(self, qnn_name, num_inputs, num_params, result):
        """Test that the algorithm results match the original code's."""
```
Unpack also needs an import.